src/platform/src/main/java/org/geoserver/platform/resource/FileSystemResourceStore.java

```diff
@@ -91,7 +97,14 @@ public FileSystemResourceStore(File resourceDirectory) {
     
     public synchronized void addListener(File file, String path, ResourceListener listener) {
         if( watcher == null ){
+            // Would be better if we could make FileSystemResource a DisposableBean instead
+            Disposer disposer = GeoServerExtensions.bean(Disposer.class);
```

Comment:
This code block is repeated over and over (lookup, register and log...) maybe make it a static method in Disposer?